functions/Test-DbaCompression.ps1

```diff
@@ -0,0 +1,371 @@
+                  ELSE 'ROW' 
+                END 
+         FROM   ##tmpcompression) 
+
+UPDATE ##tmpcompression 
+SET    [compression_type_recommendation] = 
+       tcte2.[compression_type_recommendation] 
+FROM   tmp_cte2 tcte2, 
+       ##tmpcompression tcomp2 
+WHERE  tcte2.table_name = tcomp2.table_name 
+       AND tcte2.[schema] = tcomp2.[schema] 
+       AND tcte2.index_id = tcomp2.index_id; 
+
+SET NOCOUNT ON;
+DECLARE @UpTime VARCHAR(12), @StartDate DATETIME, @sqlmajorver int, @sqlcmd NVARCHAR(500), @params NVARCHAR(500)
+SELECT @sqlmajorver = CONVERT(int, (@@microsoftversion / 0x1000000) & 0xff);
+
+IF @sqlmajorver = 9
+BEGIN
+	SET @sqlcmd = N'SELECT @StartDateOUT = login_time, @UpTimeOUT = DATEDIFF(mi, login_time, GETDATE()) FROM master..sysprocesses WHERE spid = 1';
+END
+ELSE
+BEGIN
+	SET @sqlcmd = N'SELECT @StartDateOUT = sqlserver_start_time, @UpTimeOUT = DATEDIFF(mi,sqlserver_start_time,GETDATE()) FROM sys.dm_os_sys_info';
+END
+
+SET @params = N'@StartDateOUT DATETIME OUTPUT, @UpTimeOUT VARCHAR(12) OUTPUT';
+
+EXECUTE sp_executesql @sqlcmd, @params, @StartDateOUT=@StartDate OUTPUT, @UpTimeOUT=@UpTime OUTPUT;
+
+--SELECT @StartDate AS Collecting_Data_Since, CONVERT(VARCHAR(4),@UpTime/60/24) + 'd ' + CONVERT(VARCHAR(4),@UpTime/60%24) + 'h ' + CONVERT(VARCHAR(4),@UpTime%60) + 'm' AS Collecting_Data_For
+
+SELECT 
+DBName = DB_Name()
+,[Schema] 
+,[Table_Name] 
+,[Index_Name] 
+,[Partition] 
+,[Index_ID] 
+,[Index_Type] 
+,[Percent_Scan] 
+,[Percent_Update] 
+,[ROW_estimate_Pct_of_orig] 
+,[PAGE_estimate_Pct_of_orig]
+,[Compression_Type_Recommendation] 
+,size_curKB = [size_cur]
+,size_reqKB = [size_req]
+FROM ##tmpCompression;
+
+DROP TABLE ##tmpCompression
+DROP TABLE ##tmpEstimateRow
+DROP TABLE ##tmpEstimatePage;"
+	}
+	
+	process {
+		
+		foreach ($instance in $SqlInstance) {
+			try {
+				Write-Message -Level VeryVerbose -Message "Connecting to $instance" -Target $instance
+				$server = Connect-SqlInstance -SqlInstance $instance -SqlCredential $SourceSqlCredential
+			}
+			catch {
+				Stop-Function -Message "Failed to process Instance $Instance" -ErrorRecord $_ -Target $instance -Continue
+			}
+			
+            $Server.ConnectionContext.StatementTimeout = 0
+
+			#If IncludeSystemDBs is true, include systemdbs
+			#look at all databases, online/offline/accessible/inaccessible and tell user if a db can't be queried.
+			try {
+				if ($Database) {
+					$dbs = $server.Databases | Where-Object Name -In $Database
+				}
+				elseif ($IncludeSystemDBs) {
+					$dbs = $server.Databases | Where-Object Status -eq 'Normal'
+				}
+				else {
+					$dbs = $server.Databases | Where-Object { $_.status -eq 'Normal' -and $_.IsSystemObject -eq 0 }
+				}
+				
+				if (Was-Bound "ExcludeDatabase") {
+					$dbs = $dbs | Where-Object Name -NotIn $ExcludeDatabase
+				}
+			}
+			catch {
+				Stop-Function -Message "Unable to gather databases for $instance" -ErrorRecord $_ -Continue
```

Comment:
Change this around to:

```powershell
Stop-Function -Message "Unable to gather list of databases for $instance" -Target $instance -ErrorRecord $_ -Continue`
```

This utilizes the power of the function instead of including the target in the message.